panoply/datasource.py

```diff
@@ -7,9 +10,12 @@ class DataSource(events.Emitter):
     options = None
 
     # data source constructor
-    def __init__(self, source, options = {}):
-        super(DataSource, self).__init__()
-        
+    def __init__(self, source, options = {}, events = {}):
+        super(DataSource, self).__init__(events)
+
+        if 'destination' not in source:
```

Comment:
Take a look for example on the mssql data source 
https://github.com/panoplyio/source-mssql/blob/master/mssql/mssql.py#L32
It will never set the destination to the default destination: 
"{__schemaname}_{__tablename}"